### PR TITLE
feat(Observable): RxJS doesn't even lift.

### PR DESCRIPTION
**Pull request by benlesh**
Removes `lift`, `source`, and `operator` from the Observable API. These have been deprecated for more than 3 years. It's time.

#7201

### CODE REVIEW

**Comment by demensky** [comment]
image

File: src/internal/Subject.ts [hunk 1]
```diff
@@ -108,7 +100,7 @@ export class Subject<T> extends Observable<T> implements SubscriptionLike {
 
   /** @internal */
   protected _subscribe(subscriber: Subscriber<T>): Subscription {
-    this._throwIfClosed();
+    // this._throwIfClosed();
```
Comment (demensky) [inline src/internal/Subject.ts:103]: Is there any additional change planned?

```suggestion
```

File: src/internal/util/lift.ts [hunk 2]
```diff
@@ -17,16 +9,5 @@ export function hasLift(source: any): source is { lift: InstanceType<typeof Obse
 export function operate<T, R>(
   init: (liftedSource: Observable<T>, subscriber: Subscriber<R>) => (() => void) | void
 ): OperatorFunction<T, R> {
```
Comment (demensky) [inline src/internal/util/lift.ts:11]:
```suggestion
export function operate<T, R>(
  init: (source: Observable<T>, subscriber: Subscriber<R>) => (() => void) | void
): OperatorFunction<T, R> {
```

Probably JSDoc should also be updated.

File: src/internal/util/lift.ts [hunk 3]
```diff
@@ -17,16 +9,5 @@ export function hasLift(source: any): source is { lift: InstanceType<typeof Obse
...
-  return (source: Observable<T>) => {
```
Comment (demensky) [inline src/internal/util/lift.ts:20]: Well, the file name is no longer true.

Comment (benlesh) [reply src/internal/util/lift.ts:20]: I'll address this in a followup.

File: src/internal/util/lift.ts (hunk 2 above)
Comment (benlesh) [reply src/internal/util/lift.ts:11]: I'll address this in a followup.

**Comment by benlesh** [comment]
CORE TEAM: Approval from core team. No one wants to review it. Ben will just have to trust @demensky.

@kolodny said he'll try. But also that Netflix doesn't review code.

**Comment by benlesh** [comment]
I blame @jayphelps

**Comment by jayphelps** [comment]
@benlesh so do I.

File: spec/operators/delay-spec.ts [hunk 4]
```diff
@@ -237,7 +237,7 @@ describe('delay', () => {
       const result = e1.pipe(
         repeatWhen((notifications) => {
           const delayed = notifications.pipe(delay(t));
-          subscribeSpy = sinon.spy((delayed as any)['source'], 'subscribe');
+          subscribeSpy = sinon.spy(notifications as any, 'subscribe');
```
Comment (kolodny) [inline spec/operators/delay-spec.ts:240]: Small nit but I believe you can do `sinon.spy<any>(notifications)` instead

File: spec/operators/map-spec.ts [hunk 5]
```diff
@@ -279,4 +279,29 @@ describe('map', () => {
...
+    it('can operate on any Observable-shaped object', () => {
+      const source = {
+        subscribe(observer: Observer<number>) {
+          observer.next(1);
+          observer.next(2);
+          observer.next(3);
+          observer.complete();
+          return {
+            unsubscribe() {
+              // noop
```
Comment (kolodny) [inline spec/operators/map-spec.ts:293]: Do we want to verify this was called? If not can we remove this?

**Review by kolodny — APPROVED** [review]

**Pull request merged**